app/models/discovery_rule.rb

```diff
@@ -26,6 +27,10 @@ class DiscoveryRule < ApplicationRecord
   scoped_search :on => :enabled
   scoped_search :relation => :hostgroup, :on => :name, :complete_value => true, :rename => :hostgroup
 
+  def remove_extra_spaces
+     self.search = search.strip
```

Comment:
how about to fix it on the search autocompleter, so its fixed for all screens that use search?